game/views.py

```
@@ -672,22 +737,74 @@ def register_view(request):
+                    [user_to_verify.email],
                     fail_silently=False,
                     html_message=html_message
                 )
                 return redirect('verify_otp')
-            except (SMTPException, BadHeaderError, OSError):
-                # If email fails, delete the user so they can try again
-                user.delete()
-                request.session.pop('registration_user_id', None)
-                request.session.pop('registration_otp_hash', None)
+            except (SMTPException, BadHeaderError, OSError) as e:
+                logger.error(f"Failed to send OTP email: {e}")
```

Ruff (0.15.15)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)